### PR TITLE
add: creating a jump path via menu and console command

### PR DESCRIPTION
**Pull request by Vladislav4KZ**
Differences from commit by commandcobra7:
If there is already an outgoing or two-way (bidirectional) path from the source node to the destination node, the outgoing path will be removed and an outgoing jump path will be added.

### CODE REVIEW

**Comment by jeefo** [comment]
type 0/1 is meaningless maybe

```
CR_DECLARE_SCOPED_ENUM (PathConnection,
   None = -1,
   Outgoing = 0,
   Incoming,
   Bidirectional
)
```
?

**Comment by Vladislav4KZ** [comment]
@jeefo check the latest commits.
Is it good now?

File: src/graph.cpp [hunk 1]
```diff
@@ -1070,12 +1060,19 @@ void BotGraph::pathCreate (char dir) {
...
+         addPath (nodeFrom, nodeTo, distance);
+      }
       for (auto &link : m_paths[nodeFrom].links) {
          if (link.index == nodeTo && !(link.flags & PathFlag::Jump)) {
-            erasePath (); // remove the existing outgoing path to create the jump path
+            link.flags |= PathFlag::Jump;
+            m_paths[nodeFrom].radius = 0.0f;
+            msg ("Path added from %d to %d.", nodeFrom, nodeTo);
+         }
+         else if (link.index == nodeTo && link.flags & PathFlag::Jump) {
```
Comment (jeefo) [inline src/graph.cpp:1072]: pls: (link.flags & PathFlag::Jump)

**Review by jeefo — CHANGES_REQUESTED** [review]

File: src/graph.cpp (hunk 1 above)
Comment (Vladislav4KZ) [reply src/graph.cpp]: Done

**Review by jeefo — APPROVED** [review]

**Pull request merged**